px-dropdown.html

```diff
@@ -129,478 +127,537 @@
      */
-    _newTextValue: function (evt) {
-      this.set('value', evt.detail.val);
-      this.set('displayValue', evt.detail.val);
-      this.set('selectedKey', evt.detail.key);
-
-      this.fire('change');
+    toggle: function() {
+      this.$.dropdown.toggle();
+      this.$.dropdown.style.minWidth = window.getComputedStyle(this.$.button).width;
     },
     /**
-     * This function is called when we have a maximum character width.
+     * Clears all of the selections when the clear button is pressed.
      */
-    _newMaxContCharWidth: function (evt) {
-      this.set('_maxCharWidth', evt.detail.maxContCharacterWidth);
+    _clear: function(evt) {
       evt.stopPropagation();
+      this.set('selected', null);
+      this.set('selectedValues', []);
+      this.$.dropdown.close();
     },
     /**
-     * This function provide the cell size to the dropdown content.
+     * Forces the width of the dropdown to be at least as wide as the button.
      */
-    _provideCellSize: function (evt) {
-      var dropcell = this.contentAnchor ? this.contentAnchor : this.$.dropcell,
-        rect = dropcell.getBoundingClientRect();
-
-      evt.detail.pxContent.dropCellWidth = rect.width;
-      evt.detail.pxContent.dropCellHeight = rect.height;
-
-      evt.stopPropagation();
+    attached: function() {
+      this.$.dropdown.style.minWidth = window.getComputedStyle(this.$.button).width;
+      this._updateSelection();
     },
     /**
-     * This function checks whether the chevron should be visible or hidden.
+     * Calculates the class for a disabled dropdown.
      */
-    _hideChevron: function (newValue) {
-      return (!this.hideChevron);
+    _getDisabledClass: function() {
+      return this.disabled ? 'btn--disabled' : '';
     },
     /**
-     * Returns whether this dropdown is within the path.
+     * Calculates the class for a bare dropdown.
      */
-    _selfInPath: function (path) {
-      path = path || [];
-      for (var i = 0; i < path.length; i++) {
-        if (path[i] === this) {
-          return this;
-        }
-      }
+    _getClass: function() {
+      if(this.buttonStyle === 'bare') return 'btn--bare';
+      else if(this.buttonStyle === 'bare--primary') return 'btn--bare--primary';
+      else if(this.buttonStyle === 'tertiary') return 'btn--tertiary';
+      else return ''
     },
     /**
-     * Called whenever a click happens.
+     * Searches the DOM for the `boundTarget` element.
      */
-    _onCaptureClick: function (evt) {
-
-      //We're only interested in clicks:
-      // - not on ourselves
-      // - we're currently open
-      if (!this.preventCloseOnOutsideClick && !this._selfInPath(Polymer.dom(evt).path) &&
-        this.opened) {
-
-        var content = Polymer.dom(this).querySelector("px-dropdown-content");
-        this._flipOpened();
-        content.close();
-        this._reset();
-      }
+    _getElement: function(target) {
+      return document.querySelector(target);
```

Comment:
Should drive this through the polymer API to make sure it work in shady/shadow dom.